### PR TITLE
Command refactoring

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final Acqu
                                 : null
                 ).prompt(parameters.getOpenIdConnectPromptParameter().name())
                 .claims(parameters.getClaimsRequestJson())
-                .forceRefresh(!TextUtils.isEmpty(parameters.getClaimsRequestJson()))
+                .forceRefresh(parameters.getForceRefresh())
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: needs a merge from `dev` I guess , I believe these `forceRefresh` changes already went to dev

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 2]
```diff
@@ -147,13 +101,48 @@ public void run() {
...
+                                    @Override
+                                    public void run() {
+                                        command.getCallback().onError(finalException);
+                                    }
+                                });
+                            }
                         }
-                    });
+                    }else{
+                        final Object returnResult = result;
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java]: should we null check for `returnResult` here ?, Also may be worth adding a comment that we get into the else loop here for `LoadAccountCommand` and `RemoveAccountCommand`

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 3]
```diff
@@ -147,13 +101,48 @@ public void run() {
                         }
                     });
                 } else {
-                    final boolean finalResult = result;
-                    handler.post(new Runnable() {
-                        @Override
-                        public void run() {
-                            command.getCallback().onTaskCompleted(finalResult);
+                    if(result != null && result instanceof AcquireTokenResult){
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java]: nit: better to add a helper method to post the result for readability

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java (hunk 1 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: resolve.d

**Pull request merged**